docs/api-reference.md

```diff
@@ -608,4 +624,140 @@ end
+
+---
+
+### Methods
+
+#### childrenCount
+```
+childrenCount() -> number
+```
+Returns the amount of children that the current ShallowWrapper contains.
+
+---
+
+#### find
+```
+find([constraints]) -> list[ShallowWrapper]
+```
+When a dictionary of constraints is provided, the function will filter the children that do not satisfy all given constraints. Otherwise, as `getChildren` do, it returns a list of all children wrapped into ShallowWrappers.
+
+---
+
+#### findUnique
+```
+findUnique([constraints]) -> ShallowWrapper
+```
+Similar to `find`, this method will assert that only one child satisfies the given constraints, or in the case where none is provided, will assert that there is simply only one child.
+
+---
+
+#### getChildren
+```
+getChildren() -> list[ShallowWrapper]
+```
+Returns a list of all children wrapped into ShallowWrappers.
+
+---
+
+#### getInstance
+```
+getInstance() -> Instance or nil
+```
+Returns the instance object associated with the ShallowWrapper. It can return `nil` if the component wrapped by the ShallowWrapper does not render an instance, but rather another component. Here is an example:
+
+```lua
+local function CoolComponent(props)
+	return Roact.createElement("TextLabel")
+end
+
+local function MainComponent(props)
+	return Roact.createElement("Frame", {}, {
+		Cool = Roact.createElement(CoolComponent),
+	})
+end
+```
+
+If we shallow-render the `MainComponent`, the default depth will not render the CoolComponent.
+
+```lua
+local element = Roact.createElement(MainComponent)
+
+local tree = Roact.mount(element)
+
+local wrapper = tree:getShallowWrapper()
+
+print(wrapper:getInstance() == nil) -- prints false
+
+local coolWrapper = wrapper:findUnique()
+
+print(coolWrapper:getInstance() == nil) -- prints true
+```
+
+---
+
+#### matchSnapshot
+```
+matchSnapshot(identifier)
+```
+If no previous snapshot with the given identifier exists, it will create a new StringValue instance that will contain Lua code representing the current ShallowWrapper. When an existing snapshot is found (a ModuleScript named as the provided identifier), it will require the ModuleScript and load the data from it. Then, if the loaded data is different from the current ShallowWrapper, an error will be thrown.
+
+!!! note
+	As mentionned, `matchSnapshot` will create a StringValue, named like the given identifier, in which the generated lua code will be assigned to the Value property. When these values are generated in Studio during run mode, it's important to copy back the values and convert them into ModuleScripts.
+
+---
+
+#### snapshotToString
+```
+snapshotToString() -> string
+```
+Returns the string source of the snapshot. Useful for debugging purposes.
+
+---
+
+##### Constraints
+Constraints are passed through a dictionary that maps a constraint name to it's value.
+
+| name | value type | description |
+| --- | --- | --- |
+| kind | ElementKind | Filters with the ElementKind of the rendered elements |
```

Comment:
do we need API references in this file for ElementKind since we are exposing it now as public for these constraint functions?

Reply:
Yes if we don't expose ElementKind we should remove that!